docs/how-to/uc-image-creation.rst

```diff
@@ -0,0 +1,214 @@
+
+The :doc:`../how-to/uc-boot-parameters` describes the method for dynamically configuring the system for real-time processing.
+The configuration is an iterative process that is best done together with the expected workload. 
+Once satisfied with the configurations, it is time to prepare the operating system for going into production.
+
+This guide shows how to statically set the desired Kernel command-line options for the Ubuntu Core system.
+
+To do this, we need to create a custom gadget snap, create a model assertion, and then build the OS image.
+
+
+Create the gadget snap
+----------------------
+
+The `gadget snap`_ documentation is a recommended read before starting.
+
+This is best done by forking an existing reference gadget, then changing it for our purpose.
+For example, there is the `pc gadget`_ which is suitable for most AMD64 platforms, and the `pi gadget`_ which is meant for Raspberry Pis.
+
+Let's create a custom gadget snap based on the core22 pc gadget.
+
+Clone the specific branch and enter the directory:
+
+.. code-block:: shell
+
+    git clone https://github.com/snapcore/pc-gadget.git --branch=22 --depth=1
+    cd pc-gadget
+
+
+Add the desired kernel command line in an array to ``kernel-cmdline.append`` in ``gadget/gadget-amd64.yaml``.
+For example:
+
+.. code-block:: yaml
+
+    kernel-cmdline:
+        append:
+            - nohz=on
+            - nohz_full=2-N
+            - irqaffinity=0-1
+
+
+Refer to :doc:`../reference/kernel-boot-parameters` for the list of supported parameters.
+
+Modify ``snapcraft.yaml`` to fit your application.
+At least, make sure to change the name and version to something distinct, for example to ``realtime-pc`` and ``example`` respectively.
+
+
+Now, build the gadget snap:
+
+.. code-block:: console
+
+    $ snapcraft --verbose
+    ...
+    Created snap package realtime-pc_example_amd64.snap
+
+
+.. tip::
+    You need to rebuild the snap every time you change the `snapcraft.yaml` file.
+
+
+Create the model assertion
+--------------------------
+
+The model assertion is a digitally signed document that describes the content of the Ubuntu Core image.
+Read the `model assertion`_ documentation before continuing.
+
+Below is an example model assertion in YAML, describing a `core22` Ubuntu Core
+image:
+
+.. literalinclude :: uc-image-creation/model.json
+   :language: json
+
+Create the model assertion in the same directory as the pc-gadget.
+
+Set the following:
+
+
+- ``authority-id``, ``brand-id`` to your developer ID
+- ``timestamp`` to a RFC3339 formatted string, withing you signing key's validity
+- ``store`` to you IoT App Store ID
+
+Change the value of ``model`` to something representative of your model.
+
+The ``snaps`` array is a list of snaps that get included in the image.
+The gadget snap has no ``channel`` and ``id``, because it isn't in the store.
+We're have built it locally and will pass it to the image builder.
```

Comment:
I'd change to something like:

> The gadget snap has no listed ``channel`` and ``id`` because the locally built one from earlier will be used instead of one from a Store, passing it directly to the image builder.